pkg/podeffect/pod_effect_test.go

```diff
@@ -0,0 +1,163 @@
+
+			fh, err := st.NewFramework(
+				registeredPlugins,
+				frameworkruntime.WithClientSet(cs),
+				frameworkruntime.WithInformerFactory(informerFactory),
+				frameworkruntime.WithSnapshotSharedLister(fakeSharedLister),
+				frameworkruntime.WithPodNominator(testutil.NewPodNominator()),
+			)
+			if err != nil {
+				t.Fatalf("fail to create framework: %s", err)
+			}
+			// initialize nominated pod by adding nominated pods into nominatedPodMap
+			for _, n := range test.nodeInfos {
+				for _, p := range n.Pods {
+					if p.Pod.Status.NominatedNodeName != "" {
+						addNominatedPod(p.Pod, n.Node().Name, fh.PreemptHandle())
+					}
+				}
+			}
+			pe, _ := New(nil, fh)
+			var gotList framework.NodeScoreList
+			plugin := pe.(framework.ScorePlugin)
+			for i, n := range test.nodeInfos {
+				score, err := plugin.Score(context.Background(), nil, nil, n.Node().Name)
+				if err != nil {
+					t.Errorf("unexpected error: %v", err)
+				}
+				gotList = append(gotList, framework.NodeScore{Name: test.nodeInfos[i].Node().Name, Score: score})
+			}
+
+			status := plugin.ScoreExtensions().NormalizeScore(context.Background(), nil, nil, gotList)
+			if !status.IsSuccess() {
+				t.Errorf("unexpected error: %v", status)
+			}
+
+			for i := range gotList {
+				if !reflect.DeepEqual(test.expectedList[i].Score, gotList[i].Score) {
+					t.Errorf("expected %#v, got %#v", test.expectedList[i].Score, gotList[i].Score)
+				}
+			}
+		})
+	}
+}
+
+func makeNodeInfo(node string, terminatingPodNumber int, nominatedPodNumber int) *framework.NodeInfo {
+	ni := framework.NewNodeInfo()
+	for i := 0; i < terminatingPodNumber; i++ {
+		podInfo := &framework.PodInfo{
+			Pod: makeTerminatingPod("tpod_" + node + strconv.Itoa(i+1)),
+		}
+		ni.Pods = append(ni.Pods, podInfo)
+	}
+	for i := 0; i < nominatedPodNumber; i++ {
+		podInfo := &framework.PodInfo{
+			Pod: makeNominatedPod("npod_"+node+strconv.Itoa(i+1), node),
+		}
+		ni.Pods = append(ni.Pods, podInfo)
+	}
+	ni.SetNode(&v1.Node{
+		ObjectMeta: metav1.ObjectMeta{Name: node},
+	})
+	return ni
+}
+
+func makeTerminatingPod(name string) *v1.Pod {
+	deletionTimestamp := metav1.Time{Time: time.Now()}
+	return &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:              name,
+			DeletionTimestamp: &deletionTimestamp,
+		},
+	}
+}
+
+func makeNominatedPod(podName string, nodeName string) *v1.Pod {
+	return &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: podName,
+			UID:  types.UID(podName),
+		},
+		Status: v1.PodStatus{
+			NominatedNodeName: nodeName,
+		},
+	}
+}
+
+func addNominatedPod(pod *v1.Pod, nodeName string, ph framework.PreemptHandle) *v1.Pod {
+	ph.AddNominatedPod(pod, nodeName)
+	return pod
+}
+
+var _ framework.SharedLister = &fakeSharedLister{}
+
+type fakeSharedLister struct {
+	nodes []*framework.NodeInfo
+}
+
+func (f *fakeSharedLister) NodeInfos() framework.NodeInfoLister {
+	return fakeframework.NodeInfoLister(f.nodes)
+}
```

Comment:
You can use `test/util/fake.go`

Reply:
test/util/fake.go provides NewFakeSharedLister() which accept *v1.Node as params, but what I need is *framework.NodeInfo because it concludes podInfo. I refer to allocatable_test.go.

Reply:
Maybe I can improve it by opening another PR. WDYT?

Reply:
Given this private fakeSharedLister is simply enough, you can keep it as is.